bundle/src/main/java/com/adobe/acs/commons/cloudservices/pwa/impl/PwaServiceWorkerConfigServlet.java

```diff
@@ -0,0 +1,157 @@
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.*;
+  
+import org.apache.sling.api.servlets.HttpConstants;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.sling.models.factory.ModelFactory;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.stream.StreamSupport;
+
+import static com.adobe.acs.commons.cloudservices.pwa.impl.Constants.*;
+
+@Component(
+        service = Servlet.class,
+        property = {
+                "sling.servlet.resourceTypes=cq:Page",
+                "sling.servlet.methods=" + HttpConstants.METHOD_GET,
+                "sling.servlet.selectors=pwa.service-worker",
+                "sling.servlet.extensions=json"
+        }
+)
+public class PwaServiceWorkerConfigServlet extends SlingSafeMethodsServlet {
+    private static final Logger log = LoggerFactory.getLogger(PwaServiceWorkerConfigServlet.class);
+
+    private static final JsonElement NO_CACHE_CSRF = getNoCacheCSRF();
+
+    @Reference
+    private ModelFactory modelFactory;
+
+ 
+    @Reference
+    private ResourceResolverFactory resourceResolverFactory;
+ 
+    @Override
+    protected final void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response) throws ServletException, IOException {
+        response.setContentType("application/json;charset=UTF-8");
+        response.setCharacterEncoding("UTF-8");
+
+ 
+        ResourceResolver serviceResourceResolver = null;
+        try {
+            serviceResourceResolver = resourceResolverFactory.getServiceResourceResolver(AUTH_INFO);
+
+            response.getWriter().write(getConfig(new ServiceUserRequest(request, serviceResourceResolver)).toString());
+        } catch (LoginException e) {
+            log.error("Could not obtain service user [ {} ]", SERVICE_NAME, e);
+            response.sendError(SlingHttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        } finally {
+            if (serviceResourceResolver != null) {
+                serviceResourceResolver.close();
+            }
+        }
+ 
+    }
+
+    private JsonObject getConfig(SlingHttpServletRequest request) throws ServletException {
+        final Configuration configuration = modelFactory.createModel(request, Configuration.class);
+        final ValueMap properties = configuration.getProperties();
+        final JsonObject json = new JsonObject();
+        if(properties !=null ){
+            final int version = properties.get(PN_VERSION, 1);
+            final String swScope = properties.get(KEY_SW_Scope, "/");
```

Comment:
Are we requiring the scope to be selected by the Config author? Should this be implicit based on where the cq:conf is? Would this be a pre-RR Mapped value (ie. JCR content path vs mapped path)?